src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,48 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supplySum = 0;
+        int buySum = 0;
+
+        // Чтение данных из файла
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                String operationType = parts[0];
+                int amount = Integer.parseInt(parts[1]);
+
+                // Увеличение суммы в зависимости от типа операции
+                if (operationType.equals("supply")) {
+                    supplySum += amount;
+                } else if (operationType.equals("buy")) {
+                    buySum += amount;
+                }
+            }
+        } catch (IOException e) {
```

Comment:
The exception message in the catch block does not accurately reflect the operation that failed. Since this catch block is associated with reading from a file, the message should indicate a failure in reading, not writing. Consider updating the message to something like "Can't read data from file".